rp_yal/utils.py

```diff
@@ -197,3 +193,50 @@ def get_first_matching_content_set(contentsets, fields):
                 return contentset["id"]
         if contentset["field_count"] == 0:
             return contentset["id"]
+
+
+def get_unique_page_seen_ids(org, msisdn):
+    params = {
+        "data__user_addr": msisdn,
+        "unique_pages": "true",
+    }
+    response = requests.get(
+        urljoin(org.contentrepo_url, "api/v2/custom/pageviews/"),
+        headers=get_contentrepo_headers(org),
+        params=params,
+    )
+    response.raise_for_status()
+
+    pages_seen = response.json()
+    return [p["page"] for p in pages_seen["results"]]
+
+
+def get_contentrepo_headers(org):
+    return {
+        "Content-Type": "application/json",
+        "Authorization": "Token {}".format(org.contentrepo_token),
+    }
+
+
+def get_contentset(org, contentset_id, msisdn):
+    pages_seen_ids = get_unique_page_seen_ids(org, msisdn)
+
+    params = {
+        "show_related": "true",
+        "show_tags": "true",
+    }
+    response = requests.get(
+        urljoin(org.contentrepo_url, f"api/v2/orderedcontent/{contentset_id}"),
+        headers=get_contentrepo_headers(org),
+        params=params,
+    )
+    response.raise_for_status()
+    contentset_data = response.json()
+
+    unseen_pages = []
+    for page in contentset_data["pages"]:
+        if page["id"] not in pages_seen_ids:
+            unseen_pages.append(page)
+    contentset_data["pages"] = unseen_pages
```

Comment:
I still need to write tests but I did test with real data and it worked.

All it does is mimic the `api/v2/orderedcontent/` contentrepo url but removes the pages the person has seen to reduce the amount of times RP needs to loop